content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+if you experience some problems with wrong URLs, you can try to switch
+on `+ProxyPreserveHost+` directive, which is switched off by default:
+
+[source]
+----
+ProxyPreserveHost On
+----
+
+===== mod_proxy with HTTPS
+
+If you'd like to run Jenkins with reverse proxy in HTTPS, one user
+reported that HTTPS needs to be terminated at Jenkins, not at the
+front-end Apache.
+See http://hudson.361315.n4.nabble.com/Hudson-behind-an-Apache-Reverse-Proxy-w-SSL-td370997.html[this
+e-mail thread] for more discussion.
+
+Note that you also may need to have access to your host via regular
+http, else the admin interface test will report a broken proxy setup.
+
+Alternatively, you can add an additional `+ProxyPassReverse+` directive
+to redirect non-SSL URLs generated by Jenkins to the SSL side.
+Assuming that your webserver is `+your.host.com+`, placing the following within
+the SSL virtual host definition will do the trick:
+
+[source]
+----
+ProxyRequests     Off
+ProxyPreserveHost On
+AllowEncodedSlashes NoDecode
+
+<Proxy http://localhost:8081/jenkins*>
+  Order deny,allow
+  Allow from all
+</Proxy>
+
+ProxyPass         /jenkins  http://localhost:8081/jenkins nocanon
+ProxyPassReverse  /jenkins  http://localhost:8081/jenkins
+ProxyPassReverse  /jenkins  http://your.host.com/jenkins
+----
+
+Yet another option is to rewrite the Location headers that contain
+non-ssl URL's generated by Jenkins.
+If you want to access Jenkins from https://www.example.com/jenkins,
+placing the following within the SSL virtual host definition also works:
+
+[source]
+----
+ProxyRequests     Off
+ProxyPreserveHost On
+ProxyPass /jenkins/ http://localhost:8081/jenkins/ nocanon
+AllowEncodedSlashes NoDecode
+
+<Location /jenkins/>
+  ProxyPassReverse /
+  Order deny,allow
+  Allow from all
+</Location>
+
+Header edit Location ^http://www.example.com/jenkins/ https://www.example.com/jenkins/
+----
+
+But it may also work fine to just use simple forwarding as above (the
+first HTTPS snippet), and add
+
+[source]
+----
+RequestHeader set X-Forwarded-Proto "https"
+RequestHeader set X-Forwarded-Port "443"
+----
+
+in the HTTPS site configuration, as the Docker demo (below) does.
+(`+X-Forwarded-Port+` is not interpreted by Jenkins prior to
+https://issues.jenkins-ci.org/browse/JENKINS-23294[JENKINS-23294] so it
+may also be desirable to configure the servlet container to specify the
+originating port.)
+
+The collection of snippets above simply don't work out of the box (July
+2014), here is a full Apache-oriented "sites-enabled" file (ex:
+"sites-enabled/example") for a dedicated Jenkins host, combining the
+ideas from snippets #1 and #3.
+This was formulated on the TurnKeyLinux Jenkins appliance (v 13.0),
+after having updated Jenkins to "1.572".
+TODO (if anyone understands how to do so): Define a more selective path
+for the <Proxy *> tag, instead of *; I currently have the impression
+that the <Proxy> section is not even needed.
```

Comment:
This looks like text that refutes the entire preceding section.  Either this paragraph is correct and the preceding text should be replaced because it does not work or this paragraph should be removed.

Having TODO text in pages we publish is usually best done with either an "in progress" marker (that the page is still evolving) or with a comment for the TODO.  In this case, the TODO appears inside the paragraph.
```suggestion
```